monkey/monkey_island/cc/ui/src/services/AuthService.js

```diff
@@ -29,16 +33,47 @@ export default class AuthService {
       .then(response => response.json())
       .then(response => {
         if(response.meta.code === 200){
-          this._removeToken();
+          this._removeToken(this.AUTH_TOKEN_NAME_IN_LOCALSTORAGE);
         }
         return response;
       });
   }
 
   authFetch = (url, options) => {
+    this._refreshAuthToken(url, options);
     return this._authFetch(url, options);
   };
 
+  _refreshAuthToken = (url, options) => {
+    this._authFetch(url, options).then(res => {
+      if(res.status == 401)
+      {
+        let refreshToken = this._getToken(this.REFRESH_TOKEN_NAME_IN_LOCALSTORAGE);
+        if (refreshToken) {
+        this._fetchTokenPairFromRefreshToken(this.REFRESH_AUTH_TOKEN_ENDPOINT, {
+          method: 'POST',
+          body: JSON.stringify({
+            [this.REFRESH_TOKEN_NAME_IN_RESPONSE]: this._getToken(this.REFRESH_TOKEN_NAME_IN_LOCALSTORAGE)
+          })
+        }).then(response => response.json().then(data => ({status: response.status, body: data})))
+          .then(object => {
+            if(object.status === 200) {
+              let authToken = this._getTokenFromResponse(this.AUTH_TOKEN_NAME_IN_RESPONSE, object.body);
+              this._setToken(this.AUTH_TOKEN_NAME_IN_LOCALSTORAGE, authToken);
+
+              let refreshToken = this._getTokenFromResponse(this.REFRESH_TOKEN_NAME_IN_RESPONSE, object.body);
+              console.log(authToken, refreshToken);
+              this._setToken(this.REFRRESH_TOKEN_NAME_IN_LOCALSTORAGE, refreshToken);
```

Comment:
How did you test this?
```suggestion
              this._setToken(this.REFRESH_TOKEN_NAME_IN_LOCALSTORAGE, refreshToken);
```

Reply:
Lowered down the expiration time. And waited a couple of times but I messed something up I guess. I will check